telegraf.go

```diff
@@ -30,18 +32,22 @@ func createDialConn(addr string) (net.Conn, error) {
 
 // ClientImpl ...
 type ClientImpl struct {
-	conn net.Conn
+	conn     net.Conn
+	database string
 }
 
 // NewClientImpl returns an initiated Telegraf client.
 //
 // Ex usage:
 //
-// > client := NewClientImpl("tcp://127.0.0.1:8094")
-func NewClientImpl(addr string) (*ClientImpl, error) {
+// > client := NewClientImpl("tcp://127.0.0.1:8094", "spu")
+func NewClientImpl(addr string, database string) (*ClientImpl, error) {
```

Comment:
I'm not sure how golang works under the hood, but does this PR make this a required parameter? Since the legacy Oris code works without this routing, I fear that this could be a breaking change. Can we have an overloaded constructor OR a default, empty, database name instead instead? Similarly, the "injection" of this tag should only occur on instances with a supplied database name. To keep it backwards compatible, we should default to no routing and not inject a database tag unless specified. That way it is up to the consumers of the library to "upgrade" (or not).

Reply:
To my knowledge Golang supports neither overloading, optional params, or default values. That was also my first thought, then I thought that updating these two spots next time they update might be easier. 

Let me look around for a better solution :)

Reply:
🤦 Golang. yay.

Reply:
ya, not golangs strength. You could have another method called `set_database` which would override the internal attribute which would default to empty string and leave the init method signature unchanged. Still not sure what this is used for though?